test/viewModels/guaranteeDetails/GuaranteeViewModelSpec.scala

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewModels.guaranteeDetails
+
+import base.SpecBase
+import generators.{Generators, GuaranteeDetailsUserAnswersGenerator}
+import models.DeclarationType
+import models.DeclarationType.Option4
+import models.guaranteeDetails.GuaranteeType
+import models.guaranteeDetails.GuaranteeType._
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import pages.guaranteeDetails._
+import pages.preTaskList._
+
+class GuaranteeViewModelSpec extends SpecBase with ScalaCheckPropertyChecks with Generators with GuaranteeDetailsUserAnswersGenerator {
+
+  "apply" - {
+
+    "must return row for each answer" - {
+      "when TIR" - {
+        "must return 1 row" in {
+          val initialAnswers = emptyUserAnswers.setValue(DeclarationTypePage, Option4)
+
+          forAll(arbitraryGuaranteeAnswers(initialAnswers, index)) {
+            answers =>
+              val result  = GuaranteeViewModel(answers, index)
+              val section = result.section
+              section.sectionTitle mustNot be(defined)
+              section.rows.length mustBe 1
+          }
+        }
+      }
+
+      "when not TIR" - {
+        "when 0,1,2,4,9 guarantee type" - {
+          "must return 4 rows" ignore {
```

Comment:
is this still meant to be ignored?

Reply:
Yeah it won't be 4 until we've built the liability amount and access code pages